### PR TITLE
Fixing crash when accessing JetResolutionObject in CondDB

### PR DESCRIPTION
**Pull request by yuanchao**
#### PR description:

For the crashes found while accessing the JetResolutionObject in CondDB, it's found that the bins and variables are empty with Definiton::getBins() and Definition::getVariables(). This causes segmentation violation when calling JetResolutionObject::getRecord(). As not seen in XML dump, the formula is obtained through the formula string. Later found that Definition::nBins() is actually check from the bin-name strings. This PR changes Definition::nVariables() accordingly. Also make new function overloading on createVector to access through name strings. Minimum changes to get around the crashes.

#### PR validation:

Tested locally with getPayLoadData.py on modified PI to make sure no seg. fault. Python binding parts are not touched.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport and no backport is expected.

### CODE REVIEW

File: CondFormats/JetMETObjects/src/JetResolutionObject.cc [hunk 1]
```diff
@@ -397,7 +420,10 @@ namespace JME {
       return 1;
 
 #ifndef STANDALONE
-    const auto* formula = m_definition.getFormula();
+    reco::FormulaEvaluator* Formula = new reco::FormulaEvaluator(m_definition.getFormulaString());
+    if (!Formula)
+      return 1;
+    auto formula = *Formula;
```
Comment (Dr15Jones) [inline CondFormats/JetMETObjects/src/JetResolutionObject.cc:426]: This is a memory leak. Better to do
```C++
 reco::FormulaEvaluator formula{m_definition.getFormulaString()};
```

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+db
 * tests pass

**Comment by perrotta** [comment]
+1

**Pull request merged**